fe1-web/src/core/network/NetworkConnection.ts

```diff
@@ -96,24 +218,46 @@ export class NetworkConnection {
     }
   }
 
+  /**
+   * Function called when the connection has been closed, either on purpose or because of an error
+   */
   private onClose(): void {
     console.info(`Closed websocket connection : ${this.address}`);
+
+    // it was not our intention to close the connection. try to re-connect
+    if (!this.closeIntent) {
+      this.failedConnectionAttempts += 1;
+      console.error(`Trying to re-establish a connection at address : ${this.address}`);
+
+      // only retry a certain number of times and add a wait before retrying
+      if (this.failedConnectionAttempts <= WEBSOCKET_CONNECTION_MAX_ATTEMPTS) {
+        setTimeout(() => {
+          this.reconnectIfNecessary();
+        }, WEBSOCKET_CONNECTION_FAILURE_TIMEOUT_MS);
+
+        // check if callback is set
+      } else if (this.onConnectionDeadCallback) {
+        // do not try to re-establish the connection, this connection seems
+        // to be broken for now and the near future
+        console.error(`Connection with ${this.address} broke for good`);
+        this.alive = false;
+        this.onConnectionDeadCallback();
+      }
+    }
```

Comment:
save an indentation, return early 😁 

```suggestion
    if (this.closeIntent) {
      return;
    }
    
    // it was not our intention to close the connection. try to re-connect
    this.failedConnectionAttempts += 1;
    console.error(`Trying to re-establish a connection at address : ${this.address}`);

    // only retry a certain number of times and add a wait before retrying
    if (this.failedConnectionAttempts <= WEBSOCKET_CONNECTION_MAX_ATTEMPTS) {
      setTimeout(() => {
        this.reconnectIfNecessary();
      }, WEBSOCKET_CONNECTION_FAILURE_TIMEOUT_MS);
      return;
    }
    
    if (this.onConnectionDeadCallback) {
      // do not try to re-establish the connection, this connection seems
      // to be broken for now and the near future
      console.error(`Connection with ${this.address} broke for good`);
      this.alive = false;
      this.onConnectionDeadCallback();
    }
    
```


which sort of brings about another question, shouldn't `console.error` and `this.alive = false` happen if `onConnectionDeadCallback` is not defined ? 

i.e. 
```typescript
    // do not try to re-establish the connection, this connection seems
    // to be broken for now and the near future
    console.error(`Connection with ${this.address} broke for good`);
    this.alive = false;

    if (this.onConnectionDeadCallback) {
      this.onConnectionDeadCallback();
    } 
```

Reply:
Oh yes, defintiely! Thank you for catching that! 😁